plugins/module_utils/systemd.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025, Marco Noce <nce.marco@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+def systemd_runner(module, command, **kwargs):
+    arg_formats = dict(
+        version=cmd_runner_fmt.as_fixed("--version"),
+        list_units=cmd_runner_fmt.as_fixed("list-units"),
+        no_pager=cmd_runner_fmt.as_fixed("--no-pager"),
+        typ=cmd_runner_fmt.as_fixed("--type"),
+        types=cmd_runner_fmt.as_func(lambda v: [] if not v else [",".join(v)]),
+        all=cmd_runner_fmt.as_fixed("--all"),
+        plain=cmd_runner_fmt.as_fixed("--plain"),
+        no_legend=cmd_runner_fmt.as_fixed("--no-legend"),
+        show=cmd_runner_fmt.as_fixed("show"),
+        p=cmd_runner_fmt.as_fixed("-p"),
+        props=cmd_runner_fmt.as_list(),
```

Comment:
The `p` arg is kinda artificial, it is actually about the `props`. In that case, it might be better off having one format:
```suggestion
        props=cmd_runner_fmt.as_func(lambda v: [] if not v else ["-p", ",".join(v)]),
```
and then ...